oauthlib/oauth2/rfc6749/endpoints/authorization.py

```diff
@@ -64,7 +64,12 @@ class AuthorizationEndpoint(BaseEndpoint):
     def __init__(self, default_response_type, default_token_type,
                  response_types):
         BaseEndpoint.__init__(self)
-        self._response_types = response_types
+        self._response_types = {}
+        # response_types are sorted internally so ordered comparison is faster/easier later
+
+        for k, v in response_types.iteritems() if sys.version_info[0] == 2 else iter(response_types.items()):
```

Comment:
Fair points.  On further rumination I think supporting randomly ordered response_type values is not a significant use case and the code will be cleaner overall if we simply look for `response_type` values as defined in the spec's registry.  Code reverting this is pushed.